tests/main_test.py

```diff
@@ -281,6 +281,56 @@ def test_invite(self):
         response = self.do_post('/group/delete', {'id': self.group['_id']['$oid']})
         self.ensure_status_code_msg(response, 200, "Group successfully deleted.")
 
+    def test_group_refresh_id(self):
+        """
+        test
+        :return:
+        """
+
+        # create a group
+        data = {
+            'name': 'test group name',
+            'desc': 'test group desc',
+        }
+        response = self.do_post('/group/create', {'data': data})
+        self.group = response.json()['data']
+        self.ensure_status_code_msg(response, 200, "Group successfully created.")
+
+        # add some extra stuff to check
+        data = {
+            'restricted': {
+                'permissions': {
+                    'admin_overrule_modify_transaction': False
+                }
+            }
+        }
+        response = self.do_post('/group/update', {'id':self.group['_id']['$oid'], 'data': data})
+        self.ensure_status_code_msg(response, 200, "Group successfully updated.")
+
+        # get new copy of group
+        response = self.do_post('/group/info', {'id': self.group['_id']['$oid']})
+        self.group = response.json()['data']
+        self.ensure_status_code_msg(response, 200, "Group successfully retrieved.")
+
+        # check persons groups
+        response = self.do_post('/group/refresh-id', {'id': self.group['_id']['$oid']})
+        self.ensure_status_code_msg(response, 200, "Group's unique identifier successfully refreshed.")
+        id = response.json()['id']
+
+        # check to make sure id's are different and fields are the same
+        response = self.do_post('/group/info', {'id': id})
+        self.ensure_status_code_msg(response, 200, "Group successfully retrieved.")
+        group = response.json()['data']
+        assert self.group['name'] == group['name']
+        assert self.group['desc'] == group['desc']
+        assert self.group['restricted']['permissions']['admin_overrule_modify_transaction'] \
+               == group['restricted']['permissions']['admin_overrule_modify_transaction']
+        assert self.group['_id'] != group['_id']
+
+        # delete the group
+        response = self.do_post('/group/delete', {'id': id})
+        self.ensure_status_code_msg(response, 200, "Group successfully deleted.")
```

Comment:
can we put these all in the teardown fixture and remove them from the end of all the functions?
For ex, if self.group exists then delete that group, etc

Reply:
those fixtures are meant to do this exact kind of cleanup after test case functions are run

Reply:
These functions are test specific, in order to properly test we need a clean group each time. Adding this to the teardown function will only delete it at the end of the tests

Reply:
cant get this to work at the moment and its slightly scope creep, if absolutely required we can make a separate issue for it